fleetctl/fleetctl.go

```diff
@@ -262,6 +262,26 @@ func main() {
 		os.Exit(2)
 	}
 
+	visited := make(map[string]struct{}, 0)
+	globalFlagset.Visit(func(f *flag.Flag) { visited[f.Name] = struct{}{} })
+
+	// if --driver is not set, but --endpoint looks like an etcd
+	// server, set the driver to etcd
+	if _, ok := visited["driver"]; !ok {
+		if _, ok = visited["endpoint"]; ok {
+			if eps := strings.Split(globalFlags.Endpoint, ","); len(eps) >= 1 {
+				if u, err := url.Parse(eps[0]); err == nil {
+					if _, port, err := net.SplitHostPort(u.Host); err == nil {
+						if port == "4001" || port == "2379" {
+							log.Debugf("Defaulting to --driver=%s as --endpoint appears to be etcd", clientDriverEtcd)
```

Comment:
Or if you wrap the check in a recover() and just assume that the Split, Parse, and SplitHostPort worked.